<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>allensdk.core package &mdash; Allen SDK 0.12.3 documentation</title>
    
    <link rel="stylesheet" href="_static/aibs_sphinx.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.12.3',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="Allen SDK 0.12.3 documentation" href="index.html" />
    <link rel="up" title="allensdk package" href="allensdk.html" />
    <link rel="next" title="allensdk.ephys package" href="allensdk.ephys.html" />
    <link rel="prev" title="allensdk.config.model.formats package" href="allensdk.config.model.formats.html" /> 
  </head>
  <body>
<script type="text/javascript" src="/external_assets/javascript/portal.js"></script>
<style>

	/* the icon font-family definition and helper classes */
	@font-face {
	  font-family: 'portal_icon_font';
	  src: url('/external_assets/stylesheets/portal_icon_font.eot?39505677');
	  src: url('/external_assets/stylesheets/portal_icon_font.eot?39505677#iefix') format('embedded-opentype'),
	       url('/external_assets/stylesheets/portal_icon_font.woff?39505677') format('woff'),
	       url('/external_assets/stylesheets/portal_icon_font.ttf?39505677') format('truetype'),
	       url('/external_assets/stylesheets/portal_icon_font.svg?39505677#portal_icon_font') format('svg');
	  font-weight: normal;
	  font-style: normal;
	}
	 
	 [class^="icon-"]:before, [class*=" icon-"]:before {
	  font-family: "portal_icon_font";
	  font-style: normal;
	  font-weight: normal;
	  speak: none;
	 
	  display: inline-block;
	  text-decoration: inherit;
	  width: 1em;
	  margin-right: .2em;
	  text-align: center;
	  /* opacity: .8; */
	 
	  /* For safety - reset parent styles, that can break glyph codes*/
	  font-variant: normal;
	  text-transform: none;
	     
	  /* fix buttons height, for twitter bootstrap */
	  line-height: 1em;
	 
	  /* Animation center compensation - margins should be symmetric */
	  /* remove if not needed */
	  margin-left: .2em;
	
	}
	 
	.icon-spin5:before { content: '\e806'; } /* '' */
	.icon-left-open-1:before { content: '\e81e'; } /* '' */
	.icon-plus-1:before { content: '\e801'; } /* '' */
	.icon-facebook:before { content: '\e816'; } /* '' */
	.icon-gplus:before { content: '\e818'; } /* '' */
	.icon-linkedin-3:before { content: '\e815'; } /* '' */
	.icon-twitter:before { content: '\e817'; } /* '' */
	.icon-youtube-play:before { content: '\e805'; } /* '' */
	.icon-plus-2:before { content: '\e809'; } /* '' */
	.icon-record-1:before { content: '\e81c'; } /* '' */
	.icon-cancel:before { content: '\e80b'; } /* '' */
	.icon-search:before { content: '\e804'; } /* '' */
	.icon-cancel-2:before { content: '\e80c'; } /* '' */
	.icon-plus-4:before { content: '\e80d'; } /* '' */
	.icon-cancel-3:before { content: '\e80e'; } /* '' */
	.icon-plus:before { content: '\e808'; } /* '' */
	.icon-down-open:before { content: '\e800'; } /* '' */
	.icon-right-open:before { content: '\e802'; } /* '' */
	.icon-up-open:before { content: '\e803'; } /* '' */
	.icon-stop:before { content: '\e813'; } /* '' */
	.icon-cancel-1:before { content: '\e820'; } /* '' */
	.icon-plus-3:before { content: '\e80a'; } /* '' */
	/* end icon font helpers */

	/* icon font spin animation */
	.animate-spin {
	  -moz-animation: spin 2s infinite linear;
	  -o-animation: spin 2s infinite linear;
	  -webkit-animation: spin 2s infinite linear;
	  animation: spin 2s infinite linear;
	  display: inline-block;
	}
	@-moz-keyframes spin {
	  0% {
	    -moz-transform: rotate(0deg);
	    -o-transform: rotate(0deg);
	    -webkit-transform: rotate(0deg);
	    transform: rotate(0deg);
	  }

	  100% {
	    -moz-transform: rotate(359deg);
	    -o-transform: rotate(359deg);
	    -webkit-transform: rotate(359deg);
	    transform: rotate(359deg);
	  }
	}
	@-webkit-keyframes spin {
	  0% {
	    -moz-transform: rotate(0deg);
	    -o-transform: rotate(0deg);
	    -webkit-transform: rotate(0deg);
	    transform: rotate(0deg);
	  }

	  100% {
	    -moz-transform: rotate(359deg);
	    -o-transform: rotate(359deg);
	    -webkit-transform: rotate(359deg);
	    transform: rotate(359deg);
	  }
	}
	@-o-keyframes spin {
	  0% {
	    -moz-transform: rotate(0deg);
	    -o-transform: rotate(0deg);
	    -webkit-transform: rotate(0deg);
	    transform: rotate(0deg);
	  }

	  100% {
	    -moz-transform: rotate(359deg);
	    -o-transform: rotate(359deg);
	    -webkit-transform: rotate(359deg);
	    transform: rotate(359deg);
	  }
	}
	@-ms-keyframes spin {
	  0% {
	    -moz-transform: rotate(0deg);
	    -o-transform: rotate(0deg);
	    -webkit-transform: rotate(0deg);
	    transform: rotate(0deg);
	  }

	  100% {
	    -moz-transform: rotate(359deg);
	    -o-transform: rotate(359deg);
	    -webkit-transform: rotate(359deg);
	    transform: rotate(359deg);
	  }
	}
	@keyframes spin {
	  0% {
	    -moz-transform: rotate(0deg);
	    -o-transform: rotate(0deg);
	    -webkit-transform: rotate(0deg);
	    transform: rotate(0deg);
	  }

	  100% {
	    -moz-transform: rotate(359deg);
	    -o-transform: rotate(359deg);
	    -webkit-transform: rotate(359deg);
	    transform: rotate(359deg);
	  }
	}
	/*  end spin animation */	

	@font-face {
	    font-family: 'bebas_neueregular';
	    src: url('/external_assets/stylesheets/bebasneue-webfont.eot');
	    src: url('/external_assets/stylesheets/bebasneue-webfont.eot?#iefix') format('embedded-opentype'),
	         url('/external_assets/stylesheets/bebasneue-webfont.woff') format('woff'),
	         url('/external_assets/stylesheets/bebasneue-webfont.ttf') format('truetype'),
	         url('/external_assets/stylesheets/bebasneue-webfont.svg#bebas_neueregular') format('svg');
	    font-weight: normal;
	    font-style: normal;
	}

	/* top chrome styles */
	#top_chrome {

		width:100%;
		height:28px;
		background-color:#163E65;
	}

	.chrome_link_block {

		color:#fff !important;	
		display:inline-block;
		height:100%;
		max-height:28px;
	}

	.chrome_link_block:hover {

		background-color:#eee;
		color:#0779BE !important;	
		cursor:pointer;
	}

	#top_chrome a {

		color:inherit !important;	
		text-decoration:none;
	}

	#top_chrome a:hover {

		color:inherit !important;	
	}

	#top_chrome_buttons {

		float:right;
		margin-right:50px;
	}

	.chrome_button {

		font-family:arial,sans-serif;
		font-weight:bold;
		display:inline-block;
		font-size:9px;
		text-align:left;
		text-decoration:none;
		margin:9px 20px 10px 20px;
	}

	.chrome_active {

		background-color:#1b7abd;
		color:#fff;
	}


	/* header element styles*/
	#header_content {

		font-family: bebas_neueregular, arial, sans-serif;
		font-size:22px;
		color:#0779be;		
		letter-spacing:0.03em;

		position:relative;
		width:calc(100% - 112px);
		min-width:800px;
		height:76px;
		border-bottom:2px solid #A5A8AB;
		margin-left:50px;
		margin-right:50px;
        z-index:9999; /* IT-9508 */
	}

	#portal_logo {

		position:absolute;
		top:20px;
		left:0px;
		width:236px;
	}

	#portal_tabs {

		position: absolute;
		bottom: 12px;
		left: 0px;
		margin-bottom: -12px;
		display: block;
	}

	.horizontal_list
	{
		margin: 0;
		padding: 0;
		list-style-type: none;
		display: inline; 		
	}

	.horizontal_list li { 

		display: inline; 	
	}	

	#portal_tab_menu_container {

		display:none;
	}

	#portal_search_input_group {

		position:absolute;
		margin: 0;
		padding: 0;
		bottom:12px;
		right:0px;
		border:1px solid #DAD9D3;
		color:#DAD9D3;
	}

	#portal_search_input_group input {

		margin: 0;
		padding: 0;
		border:none;
		width:110px;
		color:#DAD9D3;
		font-family:Helvetica,sans-serif;
		font-size:12px;
		height:28px;
		margin-left:10px;

		outline:none;
	}

	input::-ms-clear {
		display: none;
	}

	#portal_search_input_group i {

		font-size:0.8em;
		cursor:pointer;
	}

	#pSiteSearchTextInput:focus {

		color:#000;
	}

	.icon_menu_down {

		font-size: 12px;
		margin-left: 4px;
		vertical-align: text-top;
		padding-top: 4px;
		display: inline-block;
	}

	.portal_tab_item {

		background-color: #fff;
		padding: 12px;
		padding-left:20px;
		padding-right:20px;
		display: inline-block !important;		
	}

	.portal_tab_item a:hover {

		color:#fbe300;	
	}

	.portal_tab_item:hover {

		background-color: #0779be;	
		color:#fbe300 !important;	
	}	


	.portal_menu_active {

		background-color: #EEEEEE;
	}	

	.portal_menu_item_active {

		color:#fbe300 !important;	
	}

	.portal_tab_active {

		color:#a5a8ab !important;
	}

	#header_content a {		

		color:inherit !important;
	}

	#header_content a:hover {

		background-color: #0779be;	
		color:#fbe300 !important;	
	}

	#header_content ul {

		list-style: none;
		position: relative;
		top: 58px;
		display: inline-table;
	}

	#header_content ul ul {
		
		display:none;
		opacity:0;
	}

	/* the fly out menus */
	#header_content ul li:hover > ul,
	#header_content ul li.hover > ul {

		position: absolute;
		top: 48px;
		z-index:999;

		display: block;
		width: 152px;
		padding:0px;
		padding-top:20px;
		padding-bottom:10px;
		font-size: 12px;
		font-family: Arial,Sans-Serif;
		font-weight: bold;
		text-transform: uppercase;

		background-color: rgba(8,122,191,0.98);
		color: #fff !important;
		opacity: 1;

		-webkit-animation: fade 0.25s;
		animation: fade 0.25s;
	}

	@-webkit-keyframes fade {
	    from {opacity: 0}
	    to {opacity: 1}
	}
	@keyframes fade {
	    from {opacity: 0}
	    to {opacity: 1}
	}	

	#header_content ul li:hover > ul li {

		padding: 10px;
		padding-top: 0px;
		padding-left: 20px;
		display: block;
		opacity:1;
	}

	#header_content ul li:hover > ul a {

		display: block;
		opacity:1;
		white-space:nowrap;
	}

	#portal_menu_start_list {

		left: 81px;

		/* IE sauce */
		left:85px\9;

		/* Safari sauce */
		(;left:79px;);
	}

	#portal_menu_atlases_list {

		left: 290px;
		width: 210px !important;
        height: 236px;

		padding-bottom:5px !important;

		/* IE sauce */
		left: 303px\9;
        height: 225px\9;

		/* Safari sauce */
		(;left:279px;);
	}

	#portal_menu_tools_list {

		left: 500px;
		width: 172px !important;
		height: 236px;

		padding-bottom:5px !important;

		/* IE sauce */
		left: 513px\9;
		height:225px\9;

		/* Safari sauce */
		(;left:489px;);
		(;height:227px;);
	}

	/* mozilla sauce */
	@-moz-document url-prefix() {

		#header_content {

			height:107px;
		}

		#portal_menu_start_list {

			left:85px;
		}

		#portal_menu_atlases_list {

			left: 302px;		
		}

		#portal_menu_tools_list {

			left: 512px;
			height:236px;
		}
	}


	/* make it look OK on iPad also */
	@media all and (device-width: 768px) and (device-height: 1024px) and (orientation:portrait) {

		#portal_menu_start_list {

			left:81px;
		}

		#portal_menu_atlases_list {

			left: 291px;		
		}

		#portal_menu_tools_list {

			left: 501px;
			height:236px;
		}  
	}
	@media all and (device-width: 768px) and (device-height: 1024px) and (orientation:landscape) {

		#portal_menu_start_list {

			left:81px;
		}

		#portal_menu_atlases_list {

			left: 291px;		
		}

		#portal_menu_tools_list {

			left: 501px;
			height:236px;
		}	  
	}

	.portal_menu_list_heading {

		color:#9ccae5;
		border-bottom:1px solid #9ccae5;
		padding-bottom:5px !important;
	}

	.portal_menu_list_heading + li {

		padding-top:10px !important;
	}

	.portal_menu_list_heading li {

		padding-bottom:0px;
	}	

	#portal_menu_atlases_list li:last-of-type {

		padding-bottom:15px !important;
	}

	.icon-spin5 {

		color:#000;
	}
</style>
<div id="pHeader">
<div id="top_chrome">
	<div id="top_chrome_buttons">
		<div class="chrome_link_block"><a href="http://www.alleninstitute.org" target="_blank" class="chrome_button">ALLEN INSTITUTE</a></div>
		<div class="chrome_link_block chrome_active"><a href="http://www.brain-map.org" class="chrome_button">BRAIN ATLAS</a></div>
	</div>
</div>

<div id="header_content">

	<a href="http://AllenInstitute.github.io/AllenSDK">
		<img id="portal_logo" style="border:0;" src="/external_assets/images/Brain_Atlas_Logotype_SDK.png"/>
	</a>
</div>
</div>

<script>

jQuery('#portal_tab_menu').bind('touchstart', function() {

	if(window.console)
		console.log("touchstart");
});
</script>
<link rel="stylesheet" type="text/css" href="/external_assets/stylesheets/common_layout.css" />


  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="allensdk-core-package">
<h1>allensdk.core package<a class="headerlink" href="#allensdk-core-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this headline">¶</a></h2>
</div>
<div class="section" id="module-allensdk.core.brain_observatory_cache">
<span id="allensdk-core-brain-observatory-cache-module"></span><h2>allensdk.core.brain_observatory_cache module<a class="headerlink" href="#module-allensdk.core.brain_observatory_cache" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache">
<em class="property">class </em><tt class="descclassname">allensdk.core.brain_observatory_cache.</tt><tt class="descname">BrainObservatoryCache</tt><big>(</big><em>cache=True</em>, <em>manifest_file='brain_observatory_manifest.json'</em>, <em>base_uri=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="allensdk.api.html#allensdk.api.cache.Cache" title="allensdk.api.cache.Cache"><tt class="xref py py-class docutils literal"><span class="pre">allensdk.api.cache.Cache</span></tt></a></p>
<p>Cache class for storing and accessing data from the Brain Observatory.
By default, this class will cache any downloaded metadata or files in
well known locations defined in a manifest file.  This behavior can be
disabled.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>cache: boolean</strong></p>
<blockquote>
<div><p>Whether the class should save results of API queries to locations specified
in the manifest file.  Queries for files (as opposed to metadata) must have a
file location.  If caching is disabled, those locations must be specified
in the function call (e.g. get_ophys_experiment_data(file_name=&#8217;file.nwb&#8217;)).</p>
</div></blockquote>
<p><strong>manifest_file: string</strong></p>
<blockquote class="last">
<div><p>File name of the manifest to be read.  Default is &#8220;brain_observatory_manifest.json&#8221;.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
<p class="rubric">Attributes</p>
<table border="1" class="docutils">
<colgroup>
<col width="31%" />
<col width="69%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>api: BrainObservatoryApi instance</td>
<td>The object used for making API queries related to the Brain Observatory.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.CELL_SPECIMENS_KEY">
<tt class="descname">CELL_SPECIMENS_KEY</tt><em class="property"> = 'CELL_SPECIMENS'</em><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.CELL_SPECIMENS_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.EXPERIMENTS_KEY">
<tt class="descname">EXPERIMENTS_KEY</tt><em class="property"> = 'EXPERIMENTS'</em><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.EXPERIMENTS_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.EXPERIMENT_CONTAINERS_KEY">
<tt class="descname">EXPERIMENT_CONTAINERS_KEY</tt><em class="property"> = 'EXPERIMENT_CONTAINERS'</em><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.EXPERIMENT_CONTAINERS_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.EXPERIMENT_DATA_KEY">
<tt class="descname">EXPERIMENT_DATA_KEY</tt><em class="property"> = 'EXPERIMENT_DATA'</em><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.EXPERIMENT_DATA_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.STIMULUS_MAPPINGS_KEY">
<tt class="descname">STIMULUS_MAPPINGS_KEY</tt><em class="property"> = 'STIMULUS_MAPPINGS'</em><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.STIMULUS_MAPPINGS_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.build_manifest">
<tt class="descname">build_manifest</tt><big>(</big><em>file_name</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache.build_manifest"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.build_manifest" title="Permalink to this definition">¶</a></dt>
<dd><p>Construct a manifest for this Cache class and save it in a file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>File location to save the manifest.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_cre_lines">
<tt class="descname">get_all_cre_lines</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache.get_all_cre_lines"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_cre_lines" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of all cre driver lines in the data set.</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_imaging_depths">
<tt class="descname">get_all_imaging_depths</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache.get_all_imaging_depths"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_imaging_depths" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of all imaging depths in the data set.</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_reporter_lines">
<tt class="descname">get_all_reporter_lines</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache.get_all_reporter_lines"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_reporter_lines" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of all reporter lines in the data set.</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_session_types">
<tt class="descname">get_all_session_types</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache.get_all_session_types"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_session_types" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of all stimulus sessions in the data set.</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_stimuli">
<tt class="descname">get_all_stimuli</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache.get_all_stimuli"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_stimuli" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of all stimuli in the data set.</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_targeted_structures">
<tt class="descname">get_all_targeted_structures</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache.get_all_targeted_structures"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_all_targeted_structures" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of all targeted structures in the data set.</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_cell_specimens">
<tt class="descname">get_cell_specimens</tt><big>(</big><em>file_name=None</em>, <em>ids=None</em>, <em>experiment_container_ids=None</em>, <em>simple=True</em>, <em>filters=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache.get_cell_specimens"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_cell_specimens" title="Permalink to this definition">¶</a></dt>
<dd><p>Return cell specimens that have certain properies.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the cell specimens.  If file_name is None,
the file_name will be pulled out of the manifest.  If caching
is disabled, no file will be saved. Default is None.</p>
</div></blockquote>
<p><strong>ids: list</strong></p>
<blockquote>
<div><p>List of cell specimen ids.</p>
</div></blockquote>
<p><strong>experiment_container_ids: list</strong></p>
<blockquote>
<div><p>List of experiment container ids.</p>
</div></blockquote>
<p><strong>simple: boolean</strong></p>
<blockquote>
<div><p>Whether or not to simplify the dictionary properties returned by this method
to a more concise subset.</p>
</div></blockquote>
<p><strong>filters: list of dicts</strong></p>
<blockquote>
<div><p>List of filter dictionaries.  The Allen Brain Observatory web site can 
generate filters in this format to reproduce a filtered set of cells
found there.  To see what these look like, visit 
<a class="reference external" href="http://observatory.brain-map.org/visualcoding">http://observatory.brain-map.org/visualcoding</a>, perform a cell search
and apply some filters (e.g. find cells in a particular area), then 
click the &#8220;view these cells in the AllenSDK&#8221; link on the bottom-left
of the search results page.  This will take you to a page that contains
a code sample you can use to apply those same filters via this argument.
For more detail on the filter syntax, see BrainObservatoryApi.dataframe_query.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">list of dictionaries</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_experiment_containers">
<tt class="descname">get_experiment_containers</tt><big>(</big><em>file_name=None</em>, <em>ids=None</em>, <em>targeted_structures=None</em>, <em>imaging_depths=None</em>, <em>cre_lines=None</em>, <em>transgenic_lines=None</em>, <em>simple=True</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache.get_experiment_containers"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_experiment_containers" title="Permalink to this definition">¶</a></dt>
<dd><p>Get a list of experiment containers matching certain criteria.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the experiment containers.  If file_name is None,
the file_name will be pulled out of the manifest.  If caching
is disabled, no file will be saved. Default is None.</p>
</div></blockquote>
<p><strong>ids: list</strong></p>
<blockquote>
<div><p>List of experiment container ids.</p>
</div></blockquote>
<p><strong>targeted_structures: list</strong></p>
<blockquote>
<div><p>List of structure acronyms.  Must be in the list returned by
BrainObservatoryCache.get_all_targeted_structures().</p>
</div></blockquote>
<p><strong>imaging_depths: list</strong></p>
<blockquote>
<div><p>List of imaging depths.  Must be in the list returned by
BrainObservatoryCache.get_all_imaging_depths().</p>
</div></blockquote>
<p><strong>cre_lines: list</strong></p>
<blockquote>
<div><p>List of cre lines.  Must be in the list returned by
BrainObservatoryCache.get_all_cre_lines().</p>
</div></blockquote>
<p><strong>transgenic_lines: list</strong></p>
<blockquote>
<div><p>List of transgenic lines. Must be in the list returned by
BrainObservatoryCache.get_all_cre_lines() or.
BrainObservatoryCache.get_all_reporter_lines().</p>
</div></blockquote>
<p><strong>simple: boolean</strong></p>
<blockquote>
<div><p>Whether or not to simplify the dictionary properties returned by this method
to a more concise subset.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">list of dictionaries</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_ophys_experiment_data">
<tt class="descname">get_ophys_experiment_data</tt><big>(</big><em>ophys_experiment_id</em>, <em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache.get_ophys_experiment_data"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_ophys_experiment_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Download the NWB file for an ophys_experiment (if it hasn&#8217;t already been
downloaded) and return a data accessor object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the data set.  If file_name is None,
the file_name will be pulled out of the manifest.  If caching
is disabled, no file will be saved. Default is None.</p>
</div></blockquote>
<p><strong>ophys_experiment_id: integer</strong></p>
<blockquote>
<div><p>id of the ophys_experiment to retrieve</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">BrainObservatoryNwbDataSet</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_ophys_experiments">
<tt class="descname">get_ophys_experiments</tt><big>(</big><em>file_name=None</em>, <em>ids=None</em>, <em>experiment_container_ids=None</em>, <em>targeted_structures=None</em>, <em>imaging_depths=None</em>, <em>cre_lines=None</em>, <em>transgenic_lines=None</em>, <em>stimuli=None</em>, <em>session_types=None</em>, <em>simple=True</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_cache.html#BrainObservatoryCache.get_ophys_experiments"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_cache.BrainObservatoryCache.get_ophys_experiments" title="Permalink to this definition">¶</a></dt>
<dd><p>Get a list of ophys experiments matching certain criteria.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the ophys experiments.  If file_name is None,
the file_name will be pulled out of the manifest.  If caching
is disabled, no file will be saved. Default is None.</p>
</div></blockquote>
<p><strong>ids: list</strong></p>
<blockquote>
<div><p>List of ophys experiment ids.</p>
</div></blockquote>
<p><strong>experiment_container_ids: list</strong></p>
<blockquote>
<div><p>List of experiment container ids.</p>
</div></blockquote>
<p><strong>targeted_structures: list</strong></p>
<blockquote>
<div><p>List of structure acronyms.  Must be in the list returned by
BrainObservatoryCache.get_all_targeted_structures().</p>
</div></blockquote>
<p><strong>imaging_depths: list</strong></p>
<blockquote>
<div><p>List of imaging depths.  Must be in the list returned by
BrainObservatoryCache.get_all_imaging_depths().</p>
</div></blockquote>
<p><strong>cre_lines: list</strong></p>
<blockquote>
<div><p>List of cre lines.  Must be in the list returned by
BrainObservatoryCache.get_all_cre_lines().</p>
</div></blockquote>
<p><strong>transgenic_lines: list</strong></p>
<blockquote>
<div><p>List of transgenic lines. Must be in the list returned by
BrainObservatoryCache.get_all_cre_lines() or.
BrainObservatoryCache.get_all_reporter_lines().</p>
</div></blockquote>
<p><strong>stimuli: list</strong></p>
<blockquote>
<div><p>List of stimulus names.  Must be in the list returned by
BrainObservatoryCache.get_all_stimuli().</p>
</div></blockquote>
<p><strong>session_types: list</strong></p>
<blockquote>
<div><p>List of stimulus session type names.  Must be in the list returned by
BrainObservatoryCache.get_all_session_types().</p>
</div></blockquote>
<p><strong>simple: boolean</strong></p>
<blockquote>
<div><p>Whether or not to simplify the dictionary properties returned by this method
to a more concise subset.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">list of dictionaries</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-allensdk.core.brain_observatory_nwb_data_set">
<span id="allensdk-core-brain-observatory-nwb-data-set-module"></span><h2>allensdk.core.brain_observatory_nwb_data_set module<a class="headerlink" href="#module-allensdk.core.brain_observatory_nwb_data_set" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet">
<em class="property">class </em><tt class="descclassname">allensdk.core.brain_observatory_nwb_data_set.</tt><tt class="descname">BrainObservatoryNwbDataSet</tt><big>(</big><em>nwb_file</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<dl class="attribute">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.FILE_METADATA_MAPPING">
<tt class="descname">FILE_METADATA_MAPPING</tt><em class="property"> = {'ophys_experiment_id': 'general/session_id', 'session_type': 'general/session_type', 'fov': 'general/fov', 'sex': 'general/subject/sex', 'imaging_depth': 'general/optophysiology/imaging_plane_1/imaging depth', 'device_string': 'general/devices/2-photon microscope', 'session_start_time': 'session_start_time', 'indicator': 'general/optophysiology/imaging_plane_1/indicator', 'genotype': 'general/subject/genotype', 'age': 'general/subject/age', 'excitation_lambda': 'general/optophysiology/imaging_plane_1/excitation_lambda', 'generated_by': 'general/generated_by', 'experiment_container_id': 'general/experiment_container_id', 'targeted_structure': 'general/optophysiology/imaging_plane_1/location', 'specimen_name': 'general/specimen_name'}</em><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.FILE_METADATA_MAPPING" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.MOTION_CORRECTION_DATASETS">
<tt class="descname">MOTION_CORRECTION_DATASETS</tt><em class="property"> = ['MotionCorrection/2p_image_series/xy_translations', 'MotionCorrection/2p_image_series/xy_translation']</em><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.MOTION_CORRECTION_DATASETS" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.PIPELINE_DATASET">
<tt class="descname">PIPELINE_DATASET</tt><em class="property"> = 'brain_observatory_pipeline'</em><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.PIPELINE_DATASET" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.STIMULUS_TABLE_TYPES">
<tt class="descname">STIMULUS_TABLE_TYPES</tt><em class="property"> = {'abstract_feature_series': ['drifting_gratings', 'static_gratings'], 'indexed_time_series': ['natural_movie_one', 'natural_movie_two', 'natural_movie_three', 'natural_scenes', 'locally_sparse_noise']}</em><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.STIMULUS_TABLE_TYPES" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.SUPPORTED_PIPELINE_VERSION">
<tt class="descname">SUPPORTED_PIPELINE_VERSION</tt><em class="property"> = '1.0'</em><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.SUPPORTED_PIPELINE_VERSION" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_cell_specimen_ids">
<tt class="descname">get_cell_specimen_ids</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_cell_specimen_ids"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_cell_specimen_ids" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an array of cell IDs for all cells in the file</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">cell specimen IDs: list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_cell_specimen_indices">
<tt class="descname">get_cell_specimen_indices</tt><big>(</big><em>cell_specimen_ids</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_cell_specimen_indices"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_cell_specimen_indices" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a list of cell specimen ids, return their index based on their order in this file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>cell_specimen_ids: list of cell specimen ids</strong></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_corrected_fluorescence_traces">
<tt class="descname">get_corrected_fluorescence_traces</tt><big>(</big><em>cell_specimen_ids=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_corrected_fluorescence_traces"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_corrected_fluorescence_traces" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an array of neuropil-corrected fluorescence traces
for all ROIs and the timestamps for each datapoint</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>cell_specimen_ids: list or array (optional)</strong></p>
<blockquote>
<div><p>List of cell IDs to return traces for. If this is None (default)
then all are returned</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">timestamps: 2D numpy array</p>
<blockquote>
<div><p>Timestamp for each fluorescence sample</p>
</div></blockquote>
<p>traces: 2D numpy array</p>
<blockquote class="last">
<div><p>Corrected fluorescence traces for each cell</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_dff_traces">
<tt class="descname">get_dff_traces</tt><big>(</big><em>cell_specimen_ids=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_dff_traces"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_dff_traces" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an array of dF/F traces for all ROIs and
the timestamps for each datapoint</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>cell_specimen_ids: list or array (optional)</strong></p>
<blockquote>
<div><p>List of cell IDs to return data for. If this is None (default)
then all are returned</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">timestamps: 2D numpy array</p>
<blockquote>
<div><p>Timestamp for each fluorescence sample</p>
</div></blockquote>
<p>dF/F: 2D numpy array</p>
<blockquote class="last">
<div><p>dF/F values for each cell</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_fluorescence_timestamps">
<tt class="descname">get_fluorescence_timestamps</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_fluorescence_timestamps"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_fluorescence_timestamps" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an array of timestamps in seconds for the fluorescence traces</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_fluorescence_traces">
<tt class="descname">get_fluorescence_traces</tt><big>(</big><em>cell_specimen_ids=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_fluorescence_traces"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_fluorescence_traces" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an array of fluorescence traces for all ROI and
the timestamps for each datapoint</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>cell_specimen_ids: list or array (optional)</strong></p>
<blockquote>
<div><p>List of cell IDs to return traces for. If this is None (default)
then all are returned</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">timestamps: 2D numpy array</p>
<blockquote>
<div><p>Timestamp for each fluorescence sample</p>
</div></blockquote>
<p>traces: 2D numpy array</p>
<blockquote class="last">
<div><p>Fluorescence traces for each cell</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_locally_sparse_noise_stimulus_template">
<tt class="descname">get_locally_sparse_noise_stimulus_template</tt><big>(</big><em>mask_off_screen=True</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_locally_sparse_noise_stimulus_template"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_locally_sparse_noise_stimulus_template" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an array of the stimulus template for the specified stimulus.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>mask_off_screen: boolean</strong></p>
<blockquote>
<div><p>Set off-screen regions of the stimulus to LocallySparseNoise.LSN_OFF_SCREEN.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">tuple: (template, off-screen mask)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_max_projection">
<tt class="descname">get_max_projection</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_max_projection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_max_projection" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the maximum projection image for the 2P movie.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">max projection: np.ndarray</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_metadata">
<tt class="descname">get_metadata</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_metadata"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_metadata" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a dictionary of meta data associated with each
experiment, including Cre line, specimen number,
visual area imaged, imaging depth</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">metadata: dictionary</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_motion_correction">
<tt class="descname">get_motion_correction</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_motion_correction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_motion_correction" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a Panda DataFrame containing the x- and y- translation of each image used for image alignment</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_neuropil_traces">
<tt class="descname">get_neuropil_traces</tt><big>(</big><em>cell_specimen_ids=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_neuropil_traces"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_neuropil_traces" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an array of fluorescence traces for all ROIs
and the timestamps for each datapoint</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>cell_specimen_ids: list or array (optional)</strong></p>
<blockquote>
<div><p>List of cell IDs to return traces for. If this is None (default)
then all are returned</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">timestamps: 2D numpy array</p>
<blockquote>
<div><p>Timestamp for each fluorescence sample</p>
</div></blockquote>
<p>traces: 2D numpy array</p>
<blockquote class="last">
<div><p>Neuropil fluorescence traces for each cell</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_roi_ids">
<tt class="descname">get_roi_ids</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_roi_ids"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_roi_ids" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an array of IDs for all ROIs in the file</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">ROI IDs: list</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_roi_mask">
<tt class="descname">get_roi_mask</tt><big>(</big><em>cell_specimen_ids=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_roi_mask"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_roi_mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns an array of all the ROI masks</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>cell specimen IDs: list or array (optional)</strong></p>
<blockquote>
<div><p>List of cell IDs to return traces for. If this is None (default)
then all are returned</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">List of ROI_Mask objects</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_roi_mask_array">
<tt class="descname">get_roi_mask_array</tt><big>(</big><em>cell_specimen_ids=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_roi_mask_array"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_roi_mask_array" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a numpy array containing all of the ROI masks for requested cells.
If cell_specimen_ids is omitted, return all masks.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>cell_specimen_ids: list</strong></p>
<blockquote>
<div><p>List of cell specimen ids.  Default None.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">np.ndarray: NxWxH array, where N is number of cells</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_running_speed">
<tt class="descname">get_running_speed</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_running_speed"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_running_speed" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the mouse running speed in cm/s</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_session_type">
<tt class="descname">get_session_type</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_session_type"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_session_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the type of experimental session, presently one of the
following: three_session_A, three_session_B, three_session_C</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">session type: string</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_spontaneous_activity_stimulus_table">
<tt class="descname">get_spontaneous_activity_stimulus_table</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_spontaneous_activity_stimulus_table"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_spontaneous_activity_stimulus_table" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the spontaneous activity stimulus table, if it exists.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">stimulus table: pd.DataFrame</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_stimulus_table">
<tt class="descname">get_stimulus_table</tt><big>(</big><em>stimulus_name</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_stimulus_table"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_stimulus_table" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a stimulus table given a stimulus name</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_stimulus_template">
<tt class="descname">get_stimulus_template</tt><big>(</big><em>stimulus_name</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.get_stimulus_template"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.get_stimulus_template" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an array of the stimulus template for the specified stimulus.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>stimulus_name: string</strong></p>
<blockquote>
<div><p>Must be one of the strings returned by list_stimuli().</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">stimulus table: pd.DataFrame</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.list_stimuli">
<tt class="descname">list_stimuli</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.list_stimuli"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.list_stimuli" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of the stimuli presented in the experiment.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">stimuli: list of strings</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.save_analysis_arrays">
<tt class="descname">save_analysis_arrays</tt><big>(</big><em>*datasets</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.save_analysis_arrays"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.save_analysis_arrays" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.save_analysis_dataframes">
<tt class="descname">save_analysis_dataframes</tt><big>(</big><em>*tables</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#BrainObservatoryNwbDataSet.save_analysis_dataframes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.BrainObservatoryNwbDataSet.save_analysis_dataframes" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="exception">
<dt id="allensdk.core.brain_observatory_nwb_data_set.MissingStimulusException">
<em class="property">exception </em><tt class="descclassname">allensdk.core.brain_observatory_nwb_data_set.</tt><tt class="descname">MissingStimulusException</tt><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#MissingStimulusException"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.MissingStimulusException" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">exceptions.Exception</span></tt></p>
</dd></dl>

<dl class="function">
<dt id="allensdk.core.brain_observatory_nwb_data_set.align_running_speed">
<tt class="descclassname">allensdk.core.brain_observatory_nwb_data_set.</tt><tt class="descname">align_running_speed</tt><big>(</big><em>dxcm</em>, <em>dxtime</em>, <em>timestamps</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#align_running_speed"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.align_running_speed" title="Permalink to this definition">¶</a></dt>
<dd><p>If running speed timestamps differ from fluorescence
timestamps, adjust by inserting NaNs to running speed.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">tuple: dxcm, dxtime</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="allensdk.core.brain_observatory_nwb_data_set.make_display_mask">
<tt class="descclassname">allensdk.core.brain_observatory_nwb_data_set.</tt><tt class="descname">make_display_mask</tt><big>(</big><em>display_shape=(1920</em>, <em>1200)</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#make_display_mask"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.make_display_mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Build a display-shaped mask that indicates which pixels are on screen after warping the stimulus.</p>
</dd></dl>

<dl class="function">
<dt id="allensdk.core.brain_observatory_nwb_data_set.mask_stimulus_template">
<tt class="descclassname">allensdk.core.brain_observatory_nwb_data_set.</tt><tt class="descname">mask_stimulus_template</tt><big>(</big><em>template_display_coords</em>, <em>template_shape</em>, <em>display_mask=None</em>, <em>threshold=1.0</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#mask_stimulus_template"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.mask_stimulus_template" title="Permalink to this definition">¶</a></dt>
<dd><p>Build a mask for a stimulus template of a given shape and display coordinates that indicates
which part of the template is on screen after warping.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>template_display_coords: list</strong></p>
<blockquote>
<div><p>list of (x,y) display coordinates</p>
</div></blockquote>
<p><strong>template_shape: tuple</strong></p>
<blockquote>
<div><p>(width,height) of the display template</p>
</div></blockquote>
<p><strong>display_mask: np.ndarray</strong></p>
<blockquote>
<div><p>boolean 2D mask indicating which display coordinates are on screen after warping.</p>
</div></blockquote>
<p><strong>threshold: float</strong></p>
<blockquote>
<div><p>Fraction of pixels associated with a template display coordinate that should remain
on screen to count as belonging to the mask.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">tuple: (template mask, pixel fraction)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="allensdk.core.brain_observatory_nwb_data_set.warp_stimulus_coords">
<tt class="descclassname">allensdk.core.brain_observatory_nwb_data_set.</tt><tt class="descname">warp_stimulus_coords</tt><big>(</big><em>vertices</em>, <em>distance=15.0</em>, <em>mon_height_cm=32.5</em>, <em>mon_width_cm=51.0</em>, <em>mon_res=(1920</em>, <em>1200)</em>, <em>eyepoint=(0.5</em>, <em>0.5)</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/brain_observatory_nwb_data_set.html#warp_stimulus_coords"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.brain_observatory_nwb_data_set.warp_stimulus_coords" title="Permalink to this definition">¶</a></dt>
<dd><p>For a list of screen vertices, provides a corresponding list of texture coordinates.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>vertices: numpy.ndarray</strong></p>
<blockquote>
<div><p>[[x0,y0], [x1,y1], ...] A set of vertices to  convert to texture positions.</p>
</div></blockquote>
<p><strong>distance: float</strong></p>
<blockquote>
<div><p>distance from the monitor in cm.</p>
</div></blockquote>
<p><strong>mon_height_cm: float</strong></p>
<blockquote>
<div><p>monitor height in cm</p>
</div></blockquote>
<p><strong>mon_width_cm: float</strong></p>
<blockquote>
<div><p>monitor width in cm</p>
</div></blockquote>
<p><strong>mon_res: tuple</strong></p>
<blockquote>
<div><p>monitor resolution (x,y)</p>
</div></blockquote>
<p><strong>eyepoint: tuple</strong></p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">np.ndarray</p>
<blockquote class="last">
<div><p>x,y coordinates shaped like the input that describe what pixel coordinates
are displayed an the input coordinates after warping the stimulus.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-allensdk.core.cell_types_cache">
<span id="allensdk-core-cell-types-cache-module"></span><h2>allensdk.core.cell_types_cache module<a class="headerlink" href="#module-allensdk.core.cell_types_cache" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="allensdk.core.cell_types_cache.CellTypesCache">
<em class="property">class </em><tt class="descclassname">allensdk.core.cell_types_cache.</tt><tt class="descname">CellTypesCache</tt><big>(</big><em>cache=True</em>, <em>manifest_file='cell_types_manifest.json'</em>, <em>base_uri=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/cell_types_cache.html#CellTypesCache"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="allensdk.api.html#allensdk.api.cache.Cache" title="allensdk.api.cache.Cache"><tt class="xref py py-class docutils literal"><span class="pre">allensdk.api.cache.Cache</span></tt></a></p>
<p>Cache class for storing and accessing data from the Cell Types Database.
By default, this class will cache any downloaded metadata or files in
well known locations defined in a manifest file.  This behavior can be
disabled.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>cache: boolean</strong></p>
<blockquote>
<div><p>Whether the class should save results of API queries to locations specified
in the manifest file.  Queries for files (as opposed to metadata) must have a
file location.  If caching is disabled, those locations must be specified
in the function call (e.g. get_ephys_data(file_name=&#8217;file.nwb&#8217;)).</p>
</div></blockquote>
<p><strong>manifest_file: string</strong></p>
<blockquote class="last">
<div><p>File name of the manifest to be read.  Default is &#8220;cell_types_manifest.json&#8221;.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
<p class="rubric">Attributes</p>
<table border="1" class="docutils">
<colgroup>
<col width="26%" />
<col width="74%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>api: CellTypesApi instance</td>
<td>The object used for making API queries related to the Cell Types Database</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.CELLS_KEY">
<tt class="descname">CELLS_KEY</tt><em class="property"> = 'CELLS'</em><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.CELLS_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.EPHYS_DATA_KEY">
<tt class="descname">EPHYS_DATA_KEY</tt><em class="property"> = 'EPHYS_DATA'</em><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.EPHYS_DATA_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.EPHYS_FEATURES_KEY">
<tt class="descname">EPHYS_FEATURES_KEY</tt><em class="property"> = 'EPHYS_FEATURES'</em><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.EPHYS_FEATURES_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.EPHYS_SWEEPS_KEY">
<tt class="descname">EPHYS_SWEEPS_KEY</tt><em class="property"> = 'EPHYS_SWEEPS'</em><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.EPHYS_SWEEPS_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.MARKER_KEY">
<tt class="descname">MARKER_KEY</tt><em class="property"> = 'MARKER'</em><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.MARKER_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.MORPHOLOGY_FEATURES_KEY">
<tt class="descname">MORPHOLOGY_FEATURES_KEY</tt><em class="property"> = 'MORPHOLOGY_FEATURES'</em><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.MORPHOLOGY_FEATURES_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.RECONSTRUCTION_KEY">
<tt class="descname">RECONSTRUCTION_KEY</tt><em class="property"> = 'RECONSTRUCTION'</em><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.RECONSTRUCTION_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.build_manifest">
<tt class="descname">build_manifest</tt><big>(</big><em>file_name</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/cell_types_cache.html#CellTypesCache.build_manifest"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.build_manifest" title="Permalink to this definition">¶</a></dt>
<dd><p>Construct a manifest for this Cache class and save it in a file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>File location to save the manifest.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.get_all_features">
<tt class="descname">get_all_features</tt><big>(</big><em>dataframe=False</em>, <em>require_reconstruction=True</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/cell_types_cache.html#CellTypesCache.get_all_features"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.get_all_features" title="Permalink to this definition">¶</a></dt>
<dd><p>Download morphology and electrophysiology features for all cells and merge them
into a single table.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>dataframe: boolean</strong></p>
<blockquote>
<div><p>Return the output as a Pandas DataFrame.  If False, return
a list of dictionaries.</p>
</div></blockquote>
<p><strong>require_reconstruction: boolean</strong></p>
<blockquote class="last">
<div><p>Only return ephys and morphology features for cells that have
reconstructions. Default True.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.get_cells">
<tt class="descname">get_cells</tt><big>(</big><em>file_name=None</em>, <em>require_morphology=False</em>, <em>require_reconstruction=False</em>, <em>reporter_status=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/cell_types_cache.html#CellTypesCache.get_cells"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.get_cells" title="Permalink to this definition">¶</a></dt>
<dd><p>Download metadata for all cells in the database and optionally return a
subset filtered by whether or not they have a morphology or reconstruction.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the cell metadata as JSON.  If file_name is None,
the file_name will be pulled out of the manifest.  If caching
is disabled, no file will be saved. Default is None.</p>
</div></blockquote>
<p><strong>require_morphology: boolean</strong></p>
<blockquote>
<div><p>Filter out cells that have no morphological images.</p>
</div></blockquote>
<p><strong>require_reconstruction: boolean</strong></p>
<blockquote>
<div><p>Filter out cells that have no morphological reconstructions.</p>
</div></blockquote>
<p><strong>reporter_status: list</strong></p>
<blockquote class="last">
<div><p>Filter for cells that have one or more cell reporter statuses.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.get_ephys_data">
<tt class="descname">get_ephys_data</tt><big>(</big><em>specimen_id</em>, <em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/cell_types_cache.html#CellTypesCache.get_ephys_data"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.get_ephys_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Download electrophysiology traces for a single cell in the database.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>specimen_id: int</strong></p>
<blockquote>
<div><p>The ID of a cell specimen to download.</p>
</div></blockquote>
<p><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the ephys features metadata as CSV.
If file_name is None, the file_name will be pulled out of the
manifest.  If caching is disabled, no file will be saved.
Default is None.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">NwbDataSet</p>
<blockquote class="last">
<div><p>A class instance with helper methods for retrieving stimulus
and response traces out of an NWB file.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.get_ephys_features">
<tt class="descname">get_ephys_features</tt><big>(</big><em>dataframe=False</em>, <em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/cell_types_cache.html#CellTypesCache.get_ephys_features"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.get_ephys_features" title="Permalink to this definition">¶</a></dt>
<dd><p>Download electrophysiology features for all cells in the database.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the ephys features metadata as CSV.
If file_name is None, the file_name will be pulled out of the
manifest.  If caching is disabled, no file will be saved.
Default is None.</p>
</div></blockquote>
<p><strong>dataframe: boolean</strong></p>
<blockquote class="last">
<div><p>Return the output as a Pandas DataFrame.  If False, return
a list of dictionaries.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.get_ephys_sweeps">
<tt class="descname">get_ephys_sweeps</tt><big>(</big><em>specimen_id</em>, <em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/cell_types_cache.html#CellTypesCache.get_ephys_sweeps"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.get_ephys_sweeps" title="Permalink to this definition">¶</a></dt>
<dd><p>Download sweep metadata for a single cell specimen.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>specimen_id: int</strong></p>
<blockquote class="last">
<div><p>ID of a cell.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.get_morphology_features">
<tt class="descname">get_morphology_features</tt><big>(</big><em>dataframe=False</em>, <em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/cell_types_cache.html#CellTypesCache.get_morphology_features"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.get_morphology_features" title="Permalink to this definition">¶</a></dt>
<dd><p>Download morphology features for all cells with reconstructions in the database.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the ephys features metadata as CSV.
If file_name is None, the file_name will be pulled out of the
manifest.  If caching is disabled, no file will be saved.
Default is None.</p>
</div></blockquote>
<p><strong>dataframe: boolean</strong></p>
<blockquote class="last">
<div><p>Return the output as a Pandas DataFrame.  If False, return
a list of dictionaries.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.get_reconstruction">
<tt class="descname">get_reconstruction</tt><big>(</big><em>specimen_id</em>, <em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/cell_types_cache.html#CellTypesCache.get_reconstruction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.get_reconstruction" title="Permalink to this definition">¶</a></dt>
<dd><p>Download and open a reconstruction for a single cell in the database.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>specimen_id: int</strong></p>
<blockquote>
<div><p>The ID of a cell specimen to download.</p>
</div></blockquote>
<p><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the reconstruction SWC.
If file_name is None, the file_name will be pulled out of the
manifest.  If caching is disabled, no file will be saved.
Default is None.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Morphology</p>
<blockquote class="last">
<div><p>A class instance with methods for accessing morphology compartments.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.cell_types_cache.CellTypesCache.get_reconstruction_markers">
<tt class="descname">get_reconstruction_markers</tt><big>(</big><em>specimen_id</em>, <em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/cell_types_cache.html#CellTypesCache.get_reconstruction_markers"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.cell_types_cache.CellTypesCache.get_reconstruction_markers" title="Permalink to this definition">¶</a></dt>
<dd><p>Download and open a reconstruction marker file for a single cell in the database.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>specimen_id: int</strong></p>
<blockquote>
<div><p>The ID of a cell specimen to download.</p>
</div></blockquote>
<p><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the reconstruction marker.
If file_name is None, the file_name will be pulled out of the
manifest.  If caching is disabled, no file will be saved.
Default is None.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Morphology</p>
<blockquote class="last">
<div><p>A class instance with methods for accessing morphology compartments.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="allensdk.core.cell_types_cache.ReporterStatus">
<em class="property">class </em><tt class="descclassname">allensdk.core.cell_types_cache.</tt><tt class="descname">ReporterStatus</tt><a class="reference internal" href="_modules/allensdk/core/cell_types_cache.html#ReporterStatus"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.cell_types_cache.ReporterStatus" title="Permalink to this definition">¶</a></dt>
<dd><p>Valid strings for filtering by cell reporter status.</p>
<dl class="attribute">
<dt id="allensdk.core.cell_types_cache.ReporterStatus.INDETERMINATE">
<tt class="descname">INDETERMINATE</tt><em class="property"> = 'cre reporter indeterminate'</em><a class="headerlink" href="#allensdk.core.cell_types_cache.ReporterStatus.INDETERMINATE" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.cell_types_cache.ReporterStatus.NA">
<tt class="descname">NA</tt><em class="property"> = 'not applicable'</em><a class="headerlink" href="#allensdk.core.cell_types_cache.ReporterStatus.NA" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.cell_types_cache.ReporterStatus.NEGATIVE">
<tt class="descname">NEGATIVE</tt><em class="property"> = 'cre reporter negative'</em><a class="headerlink" href="#allensdk.core.cell_types_cache.ReporterStatus.NEGATIVE" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.cell_types_cache.ReporterStatus.POSITIVE">
<tt class="descname">POSITIVE</tt><em class="property"> = 'cre reporter positive'</em><a class="headerlink" href="#allensdk.core.cell_types_cache.ReporterStatus.POSITIVE" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

</div>
<div class="section" id="module-allensdk.core.dat_utilities">
<span id="allensdk-core-dat-utilities-module"></span><h2>allensdk.core.dat_utilities module<a class="headerlink" href="#module-allensdk.core.dat_utilities" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="allensdk.core.dat_utilities.DatUtilities">
<em class="property">class </em><tt class="descclassname">allensdk.core.dat_utilities.</tt><tt class="descname">DatUtilities</tt><a class="reference internal" href="_modules/allensdk/core/dat_utilities.html#DatUtilities"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.dat_utilities.DatUtilities" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<dl class="classmethod">
<dt id="allensdk.core.dat_utilities.DatUtilities.save_voltage">
<em class="property">classmethod </em><tt class="descname">save_voltage</tt><big>(</big><em>output_path</em>, <em>v</em>, <em>t</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/dat_utilities.html#DatUtilities.save_voltage"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.dat_utilities.DatUtilities.save_voltage" title="Permalink to this definition">¶</a></dt>
<dd><p>Save a single voltage output result into a simple text format.</p>
<p>The output file is one t v pair per line.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>output_path</strong> : string</p>
<blockquote>
<div><p>file name for output</p>
</div></blockquote>
<p><strong>v</strong> : numpy array</p>
<blockquote>
<div><p>voltage</p>
</div></blockquote>
<p><strong>t</strong> : numpy array</p>
<blockquote class="last">
<div><p>time</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-allensdk.core.json_utilities">
<span id="allensdk-core-json-utilities-module"></span><h2>allensdk.core.json_utilities module<a class="headerlink" href="#module-allensdk.core.json_utilities" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="allensdk.core.json_utilities.JsonComments">
<em class="property">class </em><tt class="descclassname">allensdk.core.json_utilities.</tt><tt class="descname">JsonComments</tt><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#JsonComments"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.JsonComments" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<dl class="classmethod">
<dt id="allensdk.core.json_utilities.JsonComments.read_file">
<em class="property">classmethod </em><tt class="descname">read_file</tt><big>(</big><em>file_name</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#JsonComments.read_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.JsonComments.read_file" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="classmethod">
<dt id="allensdk.core.json_utilities.JsonComments.read_string">
<em class="property">classmethod </em><tt class="descname">read_string</tt><big>(</big><em>json_string</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#JsonComments.read_string"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.JsonComments.read_string" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="classmethod">
<dt id="allensdk.core.json_utilities.JsonComments.remove_comments">
<em class="property">classmethod </em><tt class="descname">remove_comments</tt><big>(</big><em>json_string</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#JsonComments.remove_comments"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.JsonComments.remove_comments" title="Permalink to this definition">¶</a></dt>
<dd><p>Strip single and multiline javascript-style comments.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>json</strong> : string</p>
<blockquote>
<div><p>Json string with javascript-style comments.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">string</p>
<blockquote>
<div><p>Copy of the input with comments removed.</p>
</div></blockquote>
<p class="last">Note: A JSON decoder MAY accept and ignore comments.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="classmethod">
<dt id="allensdk.core.json_utilities.JsonComments.remove_multiline_comments">
<em class="property">classmethod </em><tt class="descname">remove_multiline_comments</tt><big>(</big><em>json_string</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#JsonComments.remove_multiline_comments"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.JsonComments.remove_multiline_comments" title="Permalink to this definition">¶</a></dt>
<dd><p>Rebuild input without substrings matching /<em>...</em>/.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>json_string</strong> : string</p>
<blockquote>
<div><p>may or may not contain multiline comments.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">string</p>
<blockquote class="last">
<div><p>Copy of the input without the comments.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="allensdk.core.json_utilities.json_handler">
<tt class="descclassname">allensdk.core.json_utilities.</tt><tt class="descname">json_handler</tt><big>(</big><em>obj</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#json_handler"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.json_handler" title="Permalink to this definition">¶</a></dt>
<dd><p>Used by write_json convert a few non-standard types to things that the json package can handle.</p>
</dd></dl>

<dl class="function">
<dt id="allensdk.core.json_utilities.read">
<tt class="descclassname">allensdk.core.json_utilities.</tt><tt class="descname">read</tt><big>(</big><em>file_name</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#read"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.read" title="Permalink to this definition">¶</a></dt>
<dd><p>Shortcut reading JSON from a file.</p>
</dd></dl>

<dl class="function">
<dt id="allensdk.core.json_utilities.read_url">
<tt class="descclassname">allensdk.core.json_utilities.</tt><tt class="descname">read_url</tt><big>(</big><em>url</em>, <em>method='POST'</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#read_url"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.read_url" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="allensdk.core.json_utilities.read_url_get">
<tt class="descclassname">allensdk.core.json_utilities.</tt><tt class="descname">read_url_get</tt><big>(</big><em>url</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#read_url_get"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.read_url_get" title="Permalink to this definition">¶</a></dt>
<dd><p>Transform a JSON contained in a file into an equivalent
nested python dict.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>url</strong> : string</p>
<blockquote>
<div><p>where to get the json.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">dict</p>
<blockquote>
<div><p>Python version of the input</p>
</div></blockquote>
<p>Note: if the input is a bare array or literal, for example,</p>
<p class="last">the output will be of the corresponding type.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="allensdk.core.json_utilities.read_url_post">
<tt class="descclassname">allensdk.core.json_utilities.</tt><tt class="descname">read_url_post</tt><big>(</big><em>url</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#read_url_post"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.read_url_post" title="Permalink to this definition">¶</a></dt>
<dd><p>Transform a JSON contained in a file into an equivalent
nested python dict.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>url</strong> : string</p>
<blockquote>
<div><p>where to get the json.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">dict</p>
<blockquote>
<div><p>Python version of the input</p>
</div></blockquote>
<p>Note: if the input is a bare array or literal, for example,</p>
<p class="last">the output will be of the corresponding type.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="allensdk.core.json_utilities.write">
<tt class="descclassname">allensdk.core.json_utilities.</tt><tt class="descname">write</tt><big>(</big><em>file_name</em>, <em>obj</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#write"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.write" title="Permalink to this definition">¶</a></dt>
<dd><p>Shortcut for writing JSON to a file.  This also takes care of serializing numpy and data types.</p>
</dd></dl>

<dl class="function">
<dt id="allensdk.core.json_utilities.write_string">
<tt class="descclassname">allensdk.core.json_utilities.</tt><tt class="descname">write_string</tt><big>(</big><em>obj</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/json_utilities.html#write_string"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.json_utilities.write_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Shortcut for writing JSON to a string.  This also takes care of serializing numpy and data types.</p>
</dd></dl>

</div>
<div class="section" id="module-allensdk.core.mouse_connectivity_cache">
<span id="allensdk-core-mouse-connectivity-cache-module"></span><h2>allensdk.core.mouse_connectivity_cache module<a class="headerlink" href="#module-allensdk.core.mouse_connectivity_cache" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache">
<em class="property">class </em><tt class="descclassname">allensdk.core.mouse_connectivity_cache.</tt><tt class="descname">MouseConnectivityCache</tt><big>(</big><em>resolution=None</em>, <em>cache=True</em>, <em>manifest_file='mouse_connectivity_manifest.json'</em>, <em>ccf_version=None</em>, <em>base_uri=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="allensdk.api.html#allensdk.api.cache.Cache" title="allensdk.api.cache.Cache"><tt class="xref py py-class docutils literal"><span class="pre">allensdk.api.cache.Cache</span></tt></a></p>
<p>Cache class for storing and accessing data related to the adult mouse
Connectivity Atlas.  By default, this class will cache any downloaded
metadata or files in well known locations defined in a manifest file.
This behavior can be disabled.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>resolution: int</strong></p>
<blockquote>
<div><p>Resolution of grid data to be downloaded when accessing projection volume,
the annotation volume, and the annotation volume.  Must be one of (10, 25,
50, 100).  Default is 25.</p>
</div></blockquote>
<p><strong>cache: boolean</strong></p>
<blockquote>
<div><p>Whether the class should save results of API queries to locations specified
in the manifest file.  Queries for files (as opposed to metadata) must have a
file location.  If caching is disabled, those locations must be specified
in the function call (e.g. get_projection_density(file_name=&#8217;file.nrrd&#8217;)).</p>
</div></blockquote>
<p><strong>manifest_file: string</strong></p>
<blockquote class="last">
<div><p>File name of the manifest to be read.  Default is &#8220;mouse_connectivity_manifest.json&#8221;.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
<p class="rubric">Attributes</p>
<table border="1" class="docutils">
<colgroup>
<col width="16%" />
<col width="84%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>resolution: int</td>
<td>Resolution of grid data to be downloaded when accessing projection volume, the annotation volume, and the annotation volume.  Must be one of (10, 25, 50, 100).  Default is 25.</td>
</tr>
<tr class="row-even"><td>api: MouseConnectivityApi instance</td>
<td>Used internally to make API queries.</td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.ANNOTATION_KEY">
<tt class="descname">ANNOTATION_KEY</tt><em class="property"> = 'ANNOTATION'</em><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.ANNOTATION_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.CCF_VERSION_KEY">
<tt class="descname">CCF_VERSION_KEY</tt><em class="property"> = 'CCF_VERSION'</em><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.CCF_VERSION_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.DATA_MASK_KEY">
<tt class="descname">DATA_MASK_KEY</tt><em class="property"> = 'DATA_MASK'</em><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.DATA_MASK_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.EXPERIMENTS_KEY">
<tt class="descname">EXPERIMENTS_KEY</tt><em class="property"> = 'EXPERIMENTS'</em><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.EXPERIMENTS_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.INJECTION_DENSITY_KEY">
<tt class="descname">INJECTION_DENSITY_KEY</tt><em class="property"> = 'INJECTION_DENSITY'</em><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.INJECTION_DENSITY_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.INJECTION_FRACTION_KEY">
<tt class="descname">INJECTION_FRACTION_KEY</tt><em class="property"> = 'INJECTION_FRACTION'</em><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.INJECTION_FRACTION_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.PROJECTION_DENSITY_KEY">
<tt class="descname">PROJECTION_DENSITY_KEY</tt><em class="property"> = 'PROJECTION_DENSITY'</em><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.PROJECTION_DENSITY_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.STRUCTURES_KEY">
<tt class="descname">STRUCTURES_KEY</tt><em class="property"> = 'STRUCTURES'</em><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.STRUCTURES_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.STRUCTURE_MASK_KEY">
<tt class="descname">STRUCTURE_MASK_KEY</tt><em class="property"> = 'STRUCTURE_MASK'</em><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.STRUCTURE_MASK_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.STRUCTURE_UNIONIZES_KEY">
<tt class="descname">STRUCTURE_UNIONIZES_KEY</tt><em class="property"> = 'STRUCTURE_UNIONIZES'</em><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.STRUCTURE_UNIONIZES_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.TEMPLATE_KEY">
<tt class="descname">TEMPLATE_KEY</tt><em class="property"> = 'TEMPLATE'</em><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.TEMPLATE_KEY" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.build_manifest">
<tt class="descname">build_manifest</tt><big>(</big><em>file_name</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.build_manifest"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.build_manifest" title="Permalink to this definition">¶</a></dt>
<dd><p>Construct a manifest for this Cache class and save it in a file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>File location to save the manifest.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.filter_experiments">
<tt class="descname">filter_experiments</tt><big>(</big><em>experiments</em>, <em>cre=None</em>, <em>injection_structure_ids=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.filter_experiments"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.filter_experiments" title="Permalink to this definition">¶</a></dt>
<dd><p>Take a list of experiments and filter them by cre status and injection structure.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>cre: boolean or list</strong></p>
<blockquote>
<div><p>If True, return only cre-positive experiments.  If False, return only
cre-negative experiments.  If None, return all experients. If list, return
all experiments with cre line names in the supplied list. Default None.</p>
</div></blockquote>
<p><strong>injection_structure_ids: list</strong></p>
<blockquote class="last">
<div><p>Only return experiments that were injected in the structures provided here.
If None, return all experiments.  Default None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.filter_structure_unionizes">
<tt class="descname">filter_structure_unionizes</tt><big>(</big><em>unionizes</em>, <em>is_injection=None</em>, <em>structure_ids=None</em>, <em>include_descendants=False</em>, <em>hemisphere_ids=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.filter_structure_unionizes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.filter_structure_unionizes" title="Permalink to this definition">¶</a></dt>
<dd><p>Take a list of unionzes and return a subset of records filtered by injection status, structure, and
hemisphere.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>is_injection: boolean</strong></p>
<blockquote>
<div><p>If True, only return unionize records that disregard non-injection pixels.
If False, only return unionize records that disregard injection pixels.
If None, return all records.  Default None.</p>
</div></blockquote>
<p><strong>structure_ids: list</strong></p>
<blockquote>
<div><p>Only return unionize records for a set of structures.
If None, return all records. Default None.</p>
</div></blockquote>
<p><strong>include_descendants: boolean</strong></p>
<blockquote>
<div><p>Include all descendant records for specified structures. Default False.</p>
</div></blockquote>
<p><strong>hemisphere_ids: list</strong></p>
<blockquote class="last">
<div><p>Only return unionize records that disregard pixels outside of a hemisphere.
or set of hemispheres. Left = 1, Right = 2, Both = 3.  If None, include all
records [1, 2, 3].  Default None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_annotation_volume">
<tt class="descname">get_annotation_volume</tt><big>(</big><em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_annotation_volume"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_annotation_volume" title="Permalink to this definition">¶</a></dt>
<dd><p>Read the annotation volume.  Download it first if it doesn&#8217;t exist.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>File name to store the annotation volume.  If it already exists,
it will be read from this file.  If file_name is None, the
file_name will be pulled out of the manifest.  Default is None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_data_mask">
<tt class="descname">get_data_mask</tt><big>(</big><em>experiment_id</em>, <em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_data_mask"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_data_mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Read a data mask volume for a single experiment. Download it
first if it doesn&#8217;t exist.  Data mask is a binary mask of
voxels that have valid data.  Only use valid data in analysis!</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>experiment_id: int</strong></p>
<blockquote>
<div><p>ID of the experiment to download/read.  This corresponds to
section_data_set_id in the API.</p>
</div></blockquote>
<p><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>File name to store the template volume.  If it already exists,
it will be read from this file.  If file_name is None, the
file_name will be pulled out of the manifest.  Default is None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_experiment_structure_unionizes">
<tt class="descname">get_experiment_structure_unionizes</tt><big>(</big><em>experiment_id</em>, <em>file_name=None</em>, <em>is_injection=None</em>, <em>structure_ids=None</em>, <em>include_descendants=False</em>, <em>hemisphere_ids=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_experiment_structure_unionizes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_experiment_structure_unionizes" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieve the structure unionize data for a specific experiment.  Filter by
structure, injection status, and hemisphere.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>experiment_id: int</strong></p>
<blockquote>
<div><p>ID of the experiment of interest.  Corresponds to section_data_set_id in the API.</p>
</div></blockquote>
<p><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the experiments list.  If file_name is None,
the file_name will be pulled out of the manifest.  If caching
is disabled, no file will be saved. Default is None.</p>
</div></blockquote>
<p><strong>is_injection: boolean</strong></p>
<blockquote>
<div><p>If True, only return unionize records that disregard non-injection pixels.
If False, only return unionize records that disregard injection pixels.
If None, return all records.  Default None.</p>
</div></blockquote>
<p><strong>structure_ids: list</strong></p>
<blockquote>
<div><p>Only return unionize records for a specific set of structures.
If None, return all records. Default None.</p>
</div></blockquote>
<p><strong>include_descendants: boolean</strong></p>
<blockquote>
<div><p>Include all descendant records for specified structures. Default False.</p>
</div></blockquote>
<p><strong>hemisphere_ids: list</strong></p>
<blockquote class="last">
<div><p>Only return unionize records that disregard pixels outside of a hemisphere.
or set of hemispheres. Left = 1, Right = 2, Both = 3.  If None, include all
records [1, 2, 3].  Default None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_experiments">
<tt class="descname">get_experiments</tt><big>(</big><em>dataframe=False</em>, <em>file_name=None</em>, <em>cre=None</em>, <em>injection_structure_ids=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_experiments"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_experiments" title="Permalink to this definition">¶</a></dt>
<dd><p>Read a list of experiments that match certain criteria.  If caching is enabled,
this will save the whole (unfiltered) list of experiments to a file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>dataframe: boolean</strong></p>
<blockquote>
<div><p>Return the list of experiments as a Pandas DataFrame.  If False,
return a list of dictionaries.  Default False.</p>
</div></blockquote>
<p><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to save/read the structures table.  If file_name is None,
the file_name will be pulled out of the manifest.  If caching
is disabled, no file will be saved. Default is None.</p>
</div></blockquote>
<p><strong>cre: boolean or list</strong></p>
<blockquote>
<div><p>If True, return only cre-positive experiments.  If False, return only
cre-negative experiments.  If None, return all experients. If list, return
all experiments with cre line names in the supplied list. Default None.</p>
</div></blockquote>
<p><strong>injection_structure_ids: list</strong></p>
<blockquote class="last">
<div><p>Only return experiments that were injected in the structures provided here.
If None, return all experiments.  Default None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_injection_density">
<tt class="descname">get_injection_density</tt><big>(</big><em>experiment_id</em>, <em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_injection_density"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_injection_density" title="Permalink to this definition">¶</a></dt>
<dd><p>Read an injection density volume for a single experiment. Download it
first if it doesn&#8217;t exist.  Injection density is the proportion of
projecting pixels in a grid voxel only including pixels that are
part of the injection site in [0,1].</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>experiment_id: int</strong></p>
<blockquote>
<div><p>ID of the experiment to download/read.  This corresponds to
section_data_set_id in the API.</p>
</div></blockquote>
<p><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>File name to store the template volume.  If it already exists,
it will be read from this file.  If file_name is None, the
file_name will be pulled out of the manifest.  Default is None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_injection_fraction">
<tt class="descname">get_injection_fraction</tt><big>(</big><em>experiment_id</em>, <em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_injection_fraction"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_injection_fraction" title="Permalink to this definition">¶</a></dt>
<dd><p>Read an injection fraction volume for a single experiment. Download it
first if it doesn&#8217;t exist.  Injection fraction is the proportion of
pixels in the injection site in a grid voxel in [0,1].</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>experiment_id: int</strong></p>
<blockquote>
<div><p>ID of the experiment to download/read.  This corresponds to
section_data_set_id in the API.</p>
</div></blockquote>
<p><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>File name to store the template volume.  If it already exists,
it will be read from this file.  If file_name is None, the
file_name will be pulled out of the manifest.  Default is None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_ontology">
<tt class="descname">get_ontology</tt><big>(</big><em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_ontology"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_ontology" title="Permalink to this definition">¶</a></dt>
<dd><p>Read the list of adult mouse structures and return an Ontology instance.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>File name to save/read the structures table.  If file_name is None,
the file_name will be pulled out of the manifest.  If caching
is disabled, no file will be saved. Default is None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_projection_density">
<tt class="descname">get_projection_density</tt><big>(</big><em>experiment_id</em>, <em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_projection_density"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_projection_density" title="Permalink to this definition">¶</a></dt>
<dd><p>Read a projection density volume for a single experiment.  Download it
first if it doesn&#8217;t exist.  Projection density is the proportion of
of projecting pixels in a grid voxel in [0,1].</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>experiment_id: int</strong></p>
<blockquote>
<div><p>ID of the experiment to download/read.  This corresponds to
section_data_set_id in the API.</p>
</div></blockquote>
<p><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>File name to store the template volume.  If it already exists,
it will be read from this file.  If file_name is None, the
file_name will be pulled out of the manifest.  Default is None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_projection_matrix">
<tt class="descname">get_projection_matrix</tt><big>(</big><em>experiment_ids</em>, <em>projection_structure_ids</em>, <em>hemisphere_ids=None</em>, <em>parameter='projection_volume'</em>, <em>dataframe=False</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_projection_matrix"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_projection_matrix" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_structure_mask">
<tt class="descname">get_structure_mask</tt><big>(</big><em>structure_id</em>, <em>file_name=None</em>, <em>annotation_file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_structure_mask"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_structure_mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Read a 3D numpy array shaped like the annotation volume that has non-zero values where
voxels belong to a particular structure.  This will take care of identifying substructures.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>structure_id: int</strong></p>
<blockquote>
<div><p>ID of a structure.</p>
</div></blockquote>
<p><strong>file_name: string</strong></p>
<blockquote>
<div><p>File name to store the structure mask.  If it already exists,
it will be read from this file.  If file_name is None, the
file_name will be pulled out of the manifest.  Default is None.</p>
</div></blockquote>
<p><strong>annotation_file_name: string</strong></p>
<blockquote class="last">
<div><p>File name to store the annotation volume.  If it already exists,
it will be read from this file.  If file_name is None, the
file_name will be pulled out of the manifest.  Default is None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_structure_unionizes">
<tt class="descname">get_structure_unionizes</tt><big>(</big><em>experiment_ids</em>, <em>is_injection=None</em>, <em>structure_ids=None</em>, <em>include_descendants=False</em>, <em>hemisphere_ids=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_structure_unionizes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_structure_unionizes" title="Permalink to this definition">¶</a></dt>
<dd><p>Get structure unionizes for a set of experiment IDs.  Filter the results by injection status,
structure, and hemisphere.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>experiment_ids: list</strong></p>
<blockquote>
<div><p>List of experiment IDs.  Corresponds to section_data_set_id in the API.</p>
</div></blockquote>
<p><strong>is_injection: boolean</strong></p>
<blockquote>
<div><p>If True, only return unionize records that disregard non-injection pixels.
If False, only return unionize records that disregard injection pixels.
If None, return all records.  Default None.</p>
</div></blockquote>
<p><strong>structure_ids: list</strong></p>
<blockquote>
<div><p>Only return unionize records for a specific set of structures.
If None, return all records. Default None.</p>
</div></blockquote>
<p><strong>include_descendants: boolean</strong></p>
<blockquote>
<div><p>Include all descendant records for specified structures. Default False.</p>
</div></blockquote>
<p><strong>hemisphere_ids: list</strong></p>
<blockquote class="last">
<div><p>Only return unionize records that disregard pixels outside of a hemisphere.
or set of hemispheres. Left = 1, Right = 2, Both = 3.  If None, include all
records [1, 2, 3].  Default None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_structures">
<tt class="descname">get_structures</tt><big>(</big><em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_structures"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_structures" title="Permalink to this definition">¶</a></dt>
<dd><p>Read the list of adult mouse structures and return a Pandas DataFrame.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>File name to save/read the structures table.  If file_name is None,
the file_name will be pulled out of the manifest.  If caching
is disabled, no file will be saved. Default is None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_template_volume">
<tt class="descname">get_template_volume</tt><big>(</big><em>file_name=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.get_template_volume"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.get_template_volume" title="Permalink to this definition">¶</a></dt>
<dd><p>Read the template volume.  Download it first if it doesn&#8217;t exist.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>File name to store the template volume.  If it already exists,
it will be read from this file.  If file_name is None, the
file_name will be pulled out of the manifest.  Default is None.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.make_structure_mask">
<tt class="descname">make_structure_mask</tt><big>(</big><em>structure_ids</em>, <em>annotation</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/mouse_connectivity_cache.html#MouseConnectivityCache.make_structure_mask"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.mouse_connectivity_cache.MouseConnectivityCache.make_structure_mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Look at an annotation volume and identify voxels that have values
in a list of structure ids.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>structure_ids: list</strong></p>
<blockquote>
<div><p>List of IDs to look for in the annotation volume</p>
</div></blockquote>
<p><strong>annotation: np.ndarray</strong></p>
<blockquote class="last">
<div><p>Numpy array filled with IDs.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-allensdk.core.nwb_data_set">
<span id="allensdk-core-nwb-data-set-module"></span><h2>allensdk.core.nwb_data_set module<a class="headerlink" href="#module-allensdk.core.nwb_data_set" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="allensdk.core.nwb_data_set.NwbDataSet">
<em class="property">class </em><tt class="descclassname">allensdk.core.nwb_data_set.</tt><tt class="descname">NwbDataSet</tt><big>(</big><em>file_name</em>, <em>spike_time_key=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/nwb_data_set.html#NwbDataSet"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.nwb_data_set.NwbDataSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>A very simple interface for exracting electrophysiology data
from an NWB file.</p>
<dl class="attribute">
<dt id="allensdk.core.nwb_data_set.NwbDataSet.DEPRECATED_SPIKE_TIMES">
<tt class="descname">DEPRECATED_SPIKE_TIMES</tt><em class="property"> = 'aibs_spike_times'</em><a class="headerlink" href="#allensdk.core.nwb_data_set.NwbDataSet.DEPRECATED_SPIKE_TIMES" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.nwb_data_set.NwbDataSet.SPIKE_TIMES">
<tt class="descname">SPIKE_TIMES</tt><em class="property"> = 'spike_times'</em><a class="headerlink" href="#allensdk.core.nwb_data_set.NwbDataSet.SPIKE_TIMES" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="allensdk.core.nwb_data_set.NwbDataSet.fill_sweep_responses">
<tt class="descname">fill_sweep_responses</tt><big>(</big><em>fill_value=0.0</em>, <em>sweep_numbers=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/nwb_data_set.html#NwbDataSet.fill_sweep_responses"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.nwb_data_set.NwbDataSet.fill_sweep_responses" title="Permalink to this definition">¶</a></dt>
<dd><p>Fill sweep response arrays with a single value.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>fill_value: float</strong></p>
<blockquote>
<div><p>Value used to fill sweep response array</p>
</div></blockquote>
<p><strong>sweep_numbers: list</strong></p>
<blockquote class="last">
<div><p>List of integer sweep numbers to be filled (default all sweeps)</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.nwb_data_set.NwbDataSet.get_experiment_sweep_numbers">
<tt class="descname">get_experiment_sweep_numbers</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/nwb_data_set.html#NwbDataSet.get_experiment_sweep_numbers"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.nwb_data_set.NwbDataSet.get_experiment_sweep_numbers" title="Permalink to this definition">¶</a></dt>
<dd><p>Get all of the sweep numbers for experiment epochs in the file, not including test sweeps.</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.nwb_data_set.NwbDataSet.get_spike_times">
<tt class="descname">get_spike_times</tt><big>(</big><em>sweep_number</em>, <em>key=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/nwb_data_set.html#NwbDataSet.get_spike_times"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.nwb_data_set.NwbDataSet.get_spike_times" title="Permalink to this definition">¶</a></dt>
<dd><p>Return any spike times stored in the NWB file for a sweep.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>sweep_number: int</strong></p>
<blockquote>
<div><p>index to access</p>
</div></blockquote>
<p><strong>key</strong> : string</p>
<blockquote>
<div><p>label where the spike times are stored (default NwbDataSet.SPIKE_TIMES)</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">list</p>
<blockquote class="last">
<div><p>list of spike times in seconds relative to the start of the sweep</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.nwb_data_set.NwbDataSet.get_sweep">
<tt class="descname">get_sweep</tt><big>(</big><em>sweep_number</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/nwb_data_set.html#NwbDataSet.get_sweep"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.nwb_data_set.NwbDataSet.get_sweep" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieve the stimulus, response, index_range, and sampling rate
for a particular sweep.  This method hides the NWB file&#8217;s distinction
between a &#8220;Sweep&#8221; and an &#8220;Experiment&#8221;.  An experiment is a subset of
of a sweep that excludes the initial test pulse.  It also excludes
any erroneous response data at the end of the sweep (usually for
ramp sweeps, where recording was terminated mid-stimulus).</p>
<p>Some sweeps do not have an experiment, so full data arrays are
returned.  Sweeps that have an experiment return full data arrays
(include the test pulse) with any erroneous data trimmed from the
back of the sweep.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>sweep_number: int</strong></p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">dict</p>
<blockquote class="last">
<div><p>A dictionary with &#8216;stimulus&#8217;, &#8216;response&#8217;, &#8216;index_range&#8217;, and
&#8216;sampling_rate&#8217; elements.  The index range is a 2-tuple where
the first element indicates the end of the test pulse and the
second index is the end of valid response data.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.nwb_data_set.NwbDataSet.get_sweep_metadata">
<tt class="descname">get_sweep_metadata</tt><big>(</big><em>sweep_number</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/nwb_data_set.html#NwbDataSet.get_sweep_metadata"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.nwb_data_set.NwbDataSet.get_sweep_metadata" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieve the sweep level metadata associated with each sweep.
Includes information on stimulus parameters like its name and amplitude
as well as recording quality metadata, like access resistance and
seal quality.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>sweep_number: int</strong></p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">dict</p>
<blockquote class="last">
<div><p>A dictionary with &#8216;aibs_stimulus_amplitude_pa&#8217;, &#8216;aibs_stimulus_name&#8217;,
&#8216;gain&#8217;, &#8216;initial_access_resistance&#8217;, &#8216;seal&#8217; elements.  These specific
fields are ones encoded in the original AIBS in vitro .nwb files.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.nwb_data_set.NwbDataSet.get_sweep_numbers">
<tt class="descname">get_sweep_numbers</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/nwb_data_set.html#NwbDataSet.get_sweep_numbers"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.nwb_data_set.NwbDataSet.get_sweep_numbers" title="Permalink to this definition">¶</a></dt>
<dd><p>Get all of the sweep numbers in the file, including test sweeps.</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.nwb_data_set.NwbDataSet.set_spike_times">
<tt class="descname">set_spike_times</tt><big>(</big><em>sweep_number</em>, <em>spike_times</em>, <em>key=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/nwb_data_set.html#NwbDataSet.set_spike_times"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.nwb_data_set.NwbDataSet.set_spike_times" title="Permalink to this definition">¶</a></dt>
<dd><p>Set or overwrite the spikes times for a sweep.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>sweep_number</strong> : int</p>
<blockquote>
<div><p>index to access</p>
</div></blockquote>
<p><strong>key</strong> : string</p>
<blockquote>
<div><p>where the times are stored (default NwbDataSet.SPIKE_TIME)</p>
</div></blockquote>
<p><strong>spike_times: np.array</strong></p>
<blockquote class="last">
<div><p>array of spike times in seconds</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.nwb_data_set.NwbDataSet.set_sweep">
<tt class="descname">set_sweep</tt><big>(</big><em>sweep_number</em>, <em>stimulus</em>, <em>response</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/nwb_data_set.html#NwbDataSet.set_sweep"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.nwb_data_set.NwbDataSet.set_sweep" title="Permalink to this definition">¶</a></dt>
<dd><p>Overwrite the stimulus or response of an NWB file.
If the supplied arrays are shorter than stored arrays,
they are padded with zeros to match the original data
size.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>sweep_number: int</strong></p>
<p><strong>stimulus: np.array</strong></p>
<blockquote>
<div><p>Overwrite the stimulus with this array.  If None, stimulus is unchanged.</p>
</div></blockquote>
<p><strong>response: np.array</strong></p>
<blockquote class="last">
<div><p>Overwrite the response with this array.  If None, response is unchanged.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-allensdk.core.ontology">
<span id="allensdk-core-ontology-module"></span><h2>allensdk.core.ontology module<a class="headerlink" href="#module-allensdk.core.ontology" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="allensdk.core.ontology.Ontology">
<em class="property">class </em><tt class="descclassname">allensdk.core.ontology.</tt><tt class="descname">Ontology</tt><big>(</big><em>df</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/ontology.html#Ontology"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.ontology.Ontology" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<dl class="method">
<dt id="allensdk.core.ontology.Ontology.get_child_ids">
<tt class="descname">get_child_ids</tt><big>(</big><em>structure_ids</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/ontology.html#Ontology.get_child_ids"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.ontology.Ontology.get_child_ids" title="Permalink to this definition">¶</a></dt>
<dd><p>Find the set of ids that are immediate children of one or more structures.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>structure_ids: iterable</strong></p>
<blockquote>
<div><p>Any iterable type that contains structure ids that can be cast to integers.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">set</p>
<blockquote class="last">
<div><p>Set of child structure ids</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.ontology.Ontology.get_children">
<tt class="descname">get_children</tt><big>(</big><em>structure_ids</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/ontology.html#Ontology.get_children"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.ontology.Ontology.get_children" title="Permalink to this definition">¶</a></dt>
<dd><p>Find the set of structures that are immediate children of one or more structures.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>structure_ids: iterable</strong></p>
<blockquote>
<div><p>Any iterable type that contains structure ids that can be cast to integers.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">pandas.DataFrame</p>
<blockquote class="last">
<div><p>Set of child structures</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.ontology.Ontology.get_descendant_ids">
<tt class="descname">get_descendant_ids</tt><big>(</big><em>structure_ids</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/ontology.html#Ontology.get_descendant_ids"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.ontology.Ontology.get_descendant_ids" title="Permalink to this definition">¶</a></dt>
<dd><p>Find the set of the ids of structures that are descendants of one or more structures.
The returned set will include the input structure ids.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>structure_ids: iterable</strong></p>
<blockquote>
<div><p>Any iterable type that contains structure ids that can be cast to integers.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">set</p>
<blockquote class="last">
<div><p>Set of descendant structure ids.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.ontology.Ontology.get_descendants">
<tt class="descname">get_descendants</tt><big>(</big><em>structure_ids</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/ontology.html#Ontology.get_descendants"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.ontology.Ontology.get_descendants" title="Permalink to this definition">¶</a></dt>
<dd><p>Find the set of structures that are descendants of one or more structures.
The returned set will include the input structures.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>structure_ids: iterable</strong></p>
<blockquote>
<div><p>Any iterable type that contains structure ids that can be cast to integers.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">pandas.DataFrame</p>
<blockquote class="last">
<div><p>Set of descendant structures.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.ontology.Ontology.structure_descends_from">
<tt class="descname">structure_descends_from</tt><big>(</big><em>child_id</em>, <em>parent_id</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/ontology.html#Ontology.structure_descends_from"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.ontology.Ontology.structure_descends_from" title="Permalink to this definition">¶</a></dt>
<dd><p>Return whether one structure id is a descendant of another structure id.</p>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-allensdk.core.swc">
<span id="allensdk-core-swc-module"></span><h2>allensdk.core.swc module<a class="headerlink" href="#module-allensdk.core.swc" title="Permalink to this headline">¶</a></h2>
<dl class="class">
<dt id="allensdk.core.swc.Compartment">
<em class="property">class </em><tt class="descclassname">allensdk.core.swc.</tt><tt class="descname">Compartment</tt><big>(</big><em>*args</em>, <em>**kwargs</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Compartment"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Compartment" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">dict</span></tt></p>
<p>A dictionary class storing information about a single morphology node</p>
<dl class="method">
<dt id="allensdk.core.swc.Compartment.print_node">
<tt class="descname">print_node</tt><big>(</big><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Compartment.print_node"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Compartment.print_node" title="Permalink to this definition">¶</a></dt>
<dd><p>print out compartment information with field names</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="allensdk.core.swc.Marker">
<em class="property">class </em><tt class="descclassname">allensdk.core.swc.</tt><tt class="descname">Marker</tt><big>(</big><em>*args</em>, <em>**kwargs</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Marker"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Marker" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">dict</span></tt></p>
<p>Simple dictionary class for handling reconstruction marker objects.</p>
<dl class="attribute">
<dt id="allensdk.core.swc.Marker.CUT_DENDRITE">
<tt class="descname">CUT_DENDRITE</tt><em class="property"> = 10</em><a class="headerlink" href="#allensdk.core.swc.Marker.CUT_DENDRITE" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Marker.NO_RECONSTRUCTION">
<tt class="descname">NO_RECONSTRUCTION</tt><em class="property"> = 20</em><a class="headerlink" href="#allensdk.core.swc.Marker.NO_RECONSTRUCTION" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Marker.SPACING">
<tt class="descname">SPACING</tt><em class="property"> = [0.1144, 0.1144, 0.28]</em><a class="headerlink" href="#allensdk.core.swc.Marker.SPACING" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="allensdk.core.swc.Morphology">
<em class="property">class </em><tt class="descclassname">allensdk.core.swc.</tt><tt class="descname">Morphology</tt><big>(</big><em>compartment_list=None</em>, <em>compartment_index=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <tt class="xref py py-class docutils literal"><span class="pre">object</span></tt></p>
<p>Keep track of the list of compartments in a morphology and provide
a few helper methods (soma, tree information, pruning, etc).</p>
<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.APICAL_DENDRITE">
<tt class="descname">APICAL_DENDRITE</tt><em class="property"> = 4</em><a class="headerlink" href="#allensdk.core.swc.Morphology.APICAL_DENDRITE" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.AXON">
<tt class="descname">AXON</tt><em class="property"> = 2</em><a class="headerlink" href="#allensdk.core.swc.Morphology.AXON" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.BASAL_DENDRITE">
<tt class="descname">BASAL_DENDRITE</tt><em class="property"> = 3</em><a class="headerlink" href="#allensdk.core.swc.Morphology.BASAL_DENDRITE" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.DENDRITE">
<tt class="descname">DENDRITE</tt><em class="property"> = 3</em><a class="headerlink" href="#allensdk.core.swc.Morphology.DENDRITE" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.NODE_TYPES">
<tt class="descname">NODE_TYPES</tt><em class="property"> = [1, 2, 3, 3, 4]</em><a class="headerlink" href="#allensdk.core.swc.Morphology.NODE_TYPES" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.SOMA">
<tt class="descname">SOMA</tt><em class="property"> = 1</em><a class="headerlink" href="#allensdk.core.swc.Morphology.SOMA" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.append">
<tt class="descname">append</tt><big>(</big><em>node_list</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.append"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.append" title="Permalink to this definition">¶</a></dt>
<dd><p>Add additional nodes to this Morphology. Those nodes must
originate from another morphology object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>node_list: list of Morphology nodes</strong></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.apply_affine">
<tt class="descname">apply_affine</tt><big>(</big><em>aff</em>, <em>scale=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.apply_affine"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.apply_affine" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply an affine transform to all compartments in this
morphology. Node radius is adjusted as well.</p>
<p>Format of the affine matrix is:</p>
<p>[x0 y0 z0]  [tx]
[x1 y1 z1]  [ty]
[x2 y2 z2]  [tz]</p>
<p>where the left 3x3 the matrix defines the affine rotation
and scaling, and the right column is the translation
vector.</p>
<p>The matrix must be collapsed and stored in a list as follows:</p>
<p>[x0 y0, z0, x1, y1, z1, x2, y2, z2, tx, ty, tz]</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>aff: 3x4 array of floats (python 2D list, or numpy 2D array)</strong></p>
<blockquote class="last">
<div><p>the transformation matrix</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.change_parent">
<tt class="descname">change_parent</tt><big>(</big><em>child</em>, <em>parent</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.change_parent"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.change_parent" title="Permalink to this definition">¶</a></dt>
<dd><p>Change the parent of a node. The child node is adjusted to
point to the new parent, the child is taken off of the previous
parent&#8217;s child list, and it is added to the new parent&#8217;s child list.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>child: integer or Morphology Object</strong></p>
<blockquote>
<div><p>The ID of the child node, or the child node itself</p>
</div></blockquote>
<p><strong>parent: integer or Morphology Object</strong></p>
<blockquote>
<div><p>The ID of the parent node, or the parent node itself</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">Nothing</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.children_of">
<tt class="descname">children_of</tt><big>(</big><em>seg</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.children_of"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.children_of" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a list of the children of the specified node</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>seg: integer or Morphology Object</strong></p>
<blockquote>
<div><p>The ID of the parent node, or the parent node itself</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">A list of the child morphology objects. If the ID of the parent</p>
<p class="last">node is invalid, None is returned.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.compartment_index">
<tt class="descname">compartment_index</tt><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.compartment_index"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.compartment_index" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the compartment index.  This is a property to ensure that the
compartment list and compartment index are in sync.</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.compartment_index_by_type">
<tt class="descname">compartment_index_by_type</tt><big>(</big><em>compartment_type</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.compartment_index_by_type"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.compartment_index_by_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an dictionary of compartments indexed by id that all have
a particular compartment type.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>compartment_type: int</strong></p>
<blockquote>
<div><p>Desired compartment type</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">A dictionary of Morphology Objects, indexed by ID</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.compartment_list">
<tt class="descname">compartment_list</tt><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.compartment_list"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.compartment_list" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the compartment list.  This is a property to ensure that the
compartment list and compartment index are in sync.</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.compartment_list_by_type">
<tt class="descname">compartment_list_by_type</tt><big>(</big><em>compartment_type</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.compartment_list_by_type"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.compartment_list_by_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an list of all compartments having the specified
compartment type.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>compartment_type: int</strong></p>
<blockquote>
<div><p>Desired compartment type</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">A list of of Morphology Objects</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.convert_type">
<tt class="descname">convert_type</tt><big>(</big><em>old_type</em>, <em>new_type</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.convert_type"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.convert_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts all compartments from one type to another.
Nodes of the original type are not affected so this
procedure can also be used as a merge procedure.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>old_type: enum</strong></p>
<blockquote>
<div><p>The compartment type to be changed.
Use one of the following constants: SOMA, AXON, DENDRITE,
BASAL_DENDRITE, or APICAL_DENDRITE</p>
</div></blockquote>
<p><strong>new_type: enum</strong></p>
<blockquote class="last">
<div><p>The target compartment type.
Use one of the following constants: SOMA, AXON, DENDRITE,
BASAL_DENDRITE, or APICAL_DENDRITE</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.delete_tree">
<tt class="descname">delete_tree</tt><big>(</big><em>n</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.delete_tree"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.delete_tree" title="Permalink to this definition">¶</a></dt>
<dd><p>Delete tree, and all of its compartments, from the morphology.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>n: Integer</strong></p>
<blockquote class="last">
<div><p>The tree number to delete</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.find">
<tt class="descname">find</tt><big>(</big><em>x</em>, <em>y</em>, <em>z</em>, <em>dist</em>, <em>node_type=None</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.find"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.find" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a list of Morphology Objects located within &#8216;dist&#8217;
of coordinate (x,y,z). If node_type is specified, the search
will be constrained to return only nodes of that type.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>x, y, z: float</strong></p>
<blockquote>
<div><p>The x,y,z coordinates from which to search around</p>
</div></blockquote>
<p><strong>dist: float</strong></p>
<blockquote>
<div><p>The search radius</p>
</div></blockquote>
<p><strong>node_type: enum (optional)</strong></p>
<blockquote>
<div><p>One of the following constants: SOMA, AXON, DENDRITE,
BASAL_DENDRITE or APICAL_DENDRITE</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">A list of all Morphology Objects matching the search criteria</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.node">
<tt class="descname">node</tt><big>(</big><em>n</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.node"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.node" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the morphology node having the specified ID.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>n: integer</strong></p>
<blockquote>
<div><p>ID of desired node</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">A morphology object having the specified ID, or None if such a</p>
<p class="last">node doesn&#8217;t exist</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.num_nodes">
<tt class="descname">num_nodes</tt><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.num_nodes"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.num_nodes" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the number of compartments in the morphology.</p>
</dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.num_trees">
<tt class="descname">num_trees</tt><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.num_trees"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.num_trees" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the number of trees in the morphology. A tree is
defined as everything following from a single root compartment.</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.parent_of">
<tt class="descname">parent_of</tt><big>(</big><em>seg</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.parent_of"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.parent_of" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns parent of the specified node.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>seg: integer or Morphology Object</strong></p>
<blockquote>
<div><p>The ID of the child node, or the child node itself</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">A morphology object, or None if no parent exists or if the</p>
<p class="last">specified node ID doesn&#8217;t exist</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.root">
<tt class="descname">root</tt><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.root"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.root" title="Permalink to this definition">¶</a></dt>
<dd><p>[deprecated] Returns root node of soma, if present. Use &#8216;soma&#8217; instead of &#8216;root&#8217;</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.save">
<tt class="descname">save</tt><big>(</big><em>file_name</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.save"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.save" title="Permalink to this definition">¶</a></dt>
<dd><p>Write this morphology out to an SWC file</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote class="last">
<div><p>desired name of your SWC file</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="allensdk.core.swc.Morphology.soma">
<tt class="descname">soma</tt><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.soma"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.soma" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns root node of soma, if present</p>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.sparsify">
<tt class="descname">sparsify</tt><big>(</big><em>modulo</em>, <em>compress_ids=False</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.sparsify"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.sparsify" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a new Morphology object that has a given number of non-leaf,
non-root nodes removed.  IDs can be reassigned so as to be continuous.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>modulo: int</strong></p>
<blockquote>
<div><p>keep 1 out of every modulo nodes.</p>
</div></blockquote>
<p><strong>compress_ids: boolean</strong></p>
<blockquote>
<div><p>Reassign ids so that ids are continuous (no missing id numbers).</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Morphology</p>
<blockquote class="last">
<div><p>A new morphology instance</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.strip_all_other_types">
<tt class="descname">strip_all_other_types</tt><big>(</big><em>node_type</em>, <em>keep_soma=True</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.strip_all_other_types"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.strip_all_other_types" title="Permalink to this definition">¶</a></dt>
<dd><p>Strips everything from the morphology except for the
specified type.
Parent and child relationships are updated accordingly, creating
new roots when necessary.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>node_type: enum</strong></p>
<blockquote>
<div><p>The compartment type to keep in the morphology.
Use one of the following constants: SOMA, AXON, DENDRITE,
BASAL_DENDRITE, or APICAL_DENDRITE</p>
</div></blockquote>
<p><strong>keep_soma: Boolean (optional)</strong></p>
<blockquote class="last">
<div><p>True (default) if soma nodes should remain in the
morpyhology, and False if the soma should also be stripped</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.strip_type">
<tt class="descname">strip_type</tt><big>(</big><em>node_type</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.strip_type"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.strip_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Strips all compartments of the specified type from the
morphology.
Parent and child relationships are updated accordingly, creating
new roots when necessary.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>node_type: enum</strong></p>
<blockquote class="last">
<div><p>The compartment type to strip from the morphology.
Use one of the following constants: SOMA, AXON, DENDRITE,
BASAL_DENDRITE, or APICAL_DENDRITE</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.stumpify_axon">
<tt class="descname">stumpify_axon</tt><big>(</big><em>count=10</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.stumpify_axon"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.stumpify_axon" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove all axon compartments except the first &#8216;count&#8217;
nodes, as counted from the connected axon root.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>count: Integer</strong></p>
<blockquote class="last">
<div><p>The length of the axon &#8216;stump&#8217;, in number of compartments</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.tree">
<tt class="descname">tree</tt><big>(</big><em>n</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.tree"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.tree" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a list of all Morphology Nodes within the specified
tree. A tree is defined as a fully connected graph of nodes.
Each tree has exactly one root.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>n: integer</strong></p>
<blockquote>
<div><p>ID of desired tree</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">A list of all morphology objects in the specified tree, or None</p>
<p class="last">if the tree doesn&#8217;t exist</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="allensdk.core.swc.Morphology.write">
<tt class="descname">write</tt><big>(</big><em>file_name</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#Morphology.write"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.Morphology.write" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="function">
<dt id="allensdk.core.swc.read_marker_file">
<tt class="descclassname">allensdk.core.swc.</tt><tt class="descname">read_marker_file</tt><big>(</big><em>file_name</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#read_marker_file"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.read_marker_file" title="Permalink to this definition">¶</a></dt>
<dd><p>read in a marker file and return a list of dictionaries</p>
</dd></dl>

<dl class="function">
<dt id="allensdk.core.swc.read_swc">
<tt class="descclassname">allensdk.core.swc.</tt><tt class="descname">read_swc</tt><big>(</big><em>file_name</em>, <em>columns='NOT_USED'</em>, <em>numeric_columns='NOT_USED'</em><big>)</big><a class="reference internal" href="_modules/allensdk/core/swc.html#read_swc"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#allensdk.core.swc.read_swc" title="Permalink to this definition">¶</a></dt>
<dd><p>Read in an SWC file and return a Morphology object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><p class="first"><strong>file_name: string</strong></p>
<blockquote>
<div><p>SWC file name.</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Morphology</p>
<blockquote class="last">
<div><p>A Morphology instance.</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-allensdk.core">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-allensdk.core" title="Permalink to this headline">¶</a></h2>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
<h3><a href="index.html">Contents</a></h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">Install Guide</a><ul class="simple">
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="data_resources.html">Data Resources</a><ul>
<li class="toctree-l2"><a class="reference internal" href="brain_observatory.html">Brain Observatory</a></li>
<li class="toctree-l2"><a class="reference internal" href="cell_types.html">Cell Types</a></li>
<li class="toctree-l2"><a class="reference internal" href="connectivity.html">Mouse Connectivity</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_api_client.html">API Access</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="models.html">Models</a><ul>
<li class="toctree-l2"><a class="reference internal" href="glif_models.html">Generalized LIF</a></li>
<li class="toctree-l2"><a class="reference internal" href="biophysical_models.html">Biophysical</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="examples.html">Examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="authors.html">Authors</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="allensdk.html">Source Documentation</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="allensdk.api.html">allensdk.api package</a></li>
<li class="toctree-l2"><a class="reference internal" href="allensdk.brain_observatory.html">allensdk.brain_observatory package</a></li>
<li class="toctree-l2"><a class="reference internal" href="allensdk.config.html">allensdk.config package</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="">allensdk.core package</a></li>
<li class="toctree-l2"><a class="reference internal" href="allensdk.ephys.html">allensdk.ephys package</a></li>
<li class="toctree-l2"><a class="reference internal" href="allensdk.model.html">allensdk.model package</a></li>
<li class="toctree-l2"><a class="reference internal" href="allensdk.morphology.html">allensdk.morphology package</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference external" href="https://github.com/AllenInstitute/AllenSDK">Github Profile</a></li>
</ul>

<h3> Questions </h3>
<p class="questions">
  Send any questions using the <a href="http://alleninstitute.org/contact_us/index.html">Send Us a Message</a> link below, 
  or submit your question to <a href="http://stackoverflow.com/">StackOverflow</a> using with the 'allensdk' tag.
</p>

<p class="questions">
  If you encounter any problems using the AllenSDK, please create an issue on <a href="http://github.com/alleninstitute/allensdk/issues/">Github's issue tracker</a>.
</p>

<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>


    <div class="footer">
    </div>
<script type="text/javascript" src="/external_assets/javascript/portalFooter.js"></script>

  </body>
</html>